cluster-api/pkg/controller/machineset/reconcile_test.go

```diff
@@ -205,15 +249,49 @@ func machineFromMachineSet(machineSet *v1alpha1.MachineSet, name string) *v1alph
 		Spec:       machineSet.Spec.Template.Spec,
 	}
 
-	amachine.Name = name
-	amachine.GenerateName = fmt.Sprintf("%s-", machineSet.Name)
+	m.Name = name
+	m.GenerateName = fmt.Sprintf("%s-", machineSet.Name)
+
+	return m
+}
+
+func machineFromMachineSet(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
+	m := machineWithoutOwnerRefs(machineSet, name)
+	m.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, controllerKind)}
+	return m
+}
+
+func machineWithPreservedExistingOwnerRefs(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
+	controller := false
+	existingRef := metav1.NewControllerRef(machineSet, controllerKind)
+	existingRef.UID = "NotMe"
+	existingRef.Controller = &controller
 
-	return amachine
+	m := machineWithoutOwnerRefs(machineSet, name)
+	m.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*existingRef, *metav1.NewControllerRef(machineSet, controllerKind)}
+	return m
 }
 
-func machineWithDifferentLabels(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
-	amachine := machineFromMachineSet(machineSet, name)
-	amachine.Labels = map[string]string{"foo":"bar"}
-	return amachine
+func setDifferentUID(m *v1alpha1.Machine) *v1alpha1.Machine {
```

Comment:
(nit)
This method is not clear. On first read, I thought it was updating the UID of the machine. Turns out it is updating the UID of the owner reference. Perhaps it could be more clearly named?

Reply:
renamed to setDifferentOwnerUID